lib/controllers/src/ActuatorControllerManager.cpp

```diff
@@ -0,0 +1,29 @@
+#include "ActuatorControllerManager.h"
+
+using namespace Controller;
+
+ActuatorControllerManager::ActuatorControllerManager(
+    const LookupTable::LookupTable<HWBRIDGE::CONTROL::Mode, ActuatorController *> &lut, HWBRIDGE::CONTROL::Mode active)
+    : m_lut(lut), m_mode(active), m_controller(nullptr) {
+  MBED_ASSERT(m_lut.find(active) != m_lut.end());
+  m_controller = m_lut.at(m_mode).value();
+}
+
+bool ActuatorControllerManager::switchControlMode(HWBRIDGE::CONTROL::Mode newControlType) {
+  m_lut.at(getActiveControlMode()).value()->stop();
+  if (auto temp = m_lut.at(newControlType); temp != std::nullopt) {
```

Comment:
why not just do `if (m_lut.at(newControlType) != std::nullopt)`?

Reply:
then you have to call .at() again inside the if no? that takes compute effort. might as well just store the value and then use it. if syntax looks weird, read up on if-init statements